webapi/src/main/scala/org/knora/webapi/responders/admin/ProjectsResponderADM.scala

```diff
@@ -436,18 +453,47 @@ final case class ProjectsResponderADMLive(
    *
    * @param id  the project's identifier (IRI / shortcode / shortname / UUID)
    *
-   * @return [[ProjectRestrictedViewSettingsGetResponseADM]]
+   * @return [[ProjectRestrictedViewSettingsResponseADM]]
    */
   override def projectRestrictedViewSettingsGetRequestADM(
     id: ProjectIdentifierADM
-  ): Task[ProjectRestrictedViewSettingsGetResponseADM] =
+  ): Task[ProjectRestrictedViewSettingsResponseADM] =
     projectRestrictedViewSettingsGetADM(id)
       .flatMap(ZIO.fromOption(_))
       .mapBoth(
         _ => NotFoundException(s"Project '${getId(id)}' not found."),
-        ProjectRestrictedViewSettingsGetResponseADM
+        ProjectRestrictedViewSettingsResponseADM
       )
 
+  /**
+   * Sets project's restricted view settings.
+   *
+   * @param iri the project's iri,
+   * @param user  requesting user,
+   * @param size  value to be set,
+   * @return [[ProjectRestrictedViewSettingsResponseADM]].
+   */
+  override def setProjectRestrictedViewSettings(
+    iri: Iri.ProjectIri,
+    user: UserADM,
+    size: RestrictedViewSize
+  ): Task[ProjectRestrictedViewSizeResponseADM] = {
+    val id = ProjectIdentifierADM.IriIdentifier(iri)
+    for {
+      _ <- getProjectFromCacheOrTriplestore(id)
+             .flatMap(ZIO.fromOption(_))
+             .orElseFail(NotFoundException(s"Project '${getId(id)}' not found."))
+      _ <- ZIO
+             .fail(ForbiddenException("User is neither system nor project admin."))
+             .unless(user.permissions.isSystemAdmin || user.permissions.isProjectAdmin(iri.value))
+
+      query = twirl.queries.sparql.admin.txt
+                .setProjectRestrictedViewSettings(iri.value, size.value, None)
+
+      _ <- triplestoreService.sparqlHttpUpdate(query.toString)
```

Comment:
Your PR has conflicts with main, please resolve. 

You will also find that the `sparqlHttpUpdate` method does not exist anymore, you need to use `query(Update("<your query>")`.